migration/idempotent/001-base.sql

```diff
@@ -2182,19 +2174,21 @@ BEGIN
             WHERE indicator IS NULL
         ), deleted_series AS (
             DELETE FROM prom_data_series.%3$I
-            WHERE delete_epoch <= %4$L
+            WHERE mark_for_deletion_epoch < %4$L
                 AND id IN (SELECT id FROM dead_series) -- concurrency means we need this qual in both
-            RETURNING id, labels
+            RETURNING id, labels, mark_for_deletion_epoch
         ), resurrected_series AS (
             UPDATE prom_data_series.%3$I
-            SET delete_epoch = NULL
-            WHERE delete_epoch <= %4$L
+            SET mark_for_deletion_epoch = NULL
+            WHERE mark_for_deletion_epoch < %4$L
                 AND id NOT IN (SELECT id FROM dead_series) -- concurrency means we need this qual in both
         )
-        SELECT ARRAY(SELECT DISTINCT unnest(labels) as label_id
-            FROM deleted_series)
-    $query$, metric_schema, metric_table, metric_series_table, deletion_epoch) INTO label_array;
-
+        SELECT
+          array_agg(DISTINCT labels.label) as label_id
+        , max(mark_for_deletion_epoch) as max_deletion_time
+        FROM deleted_series d,
+        LATERAL unnest(d.labels) as labels(label)
+    $query$, metric_schema, metric_table, metric_series_table, deletion_time) INTO label_array, max_deletion_time;
```

Comment:
Where did `max_deletion_time` come from? I don't remember it appearing either in specs or in Mat's proof.